ibm/resource_ibm_private_dns_glb_monitor.go

```diff
@@ -0,0 +1,520 @@
+	pdnsGlbMonitorHeaders       = "headers"
+	pdnsGlbMonitorHeadersName   = "name"
+	pdnsGlbMonitorHeadersValue  = "value"
+	pdnsGlbMonitorCreatedOn     = "created_on"
+	pdnsGlbMonitorModifiedOn    = "modified_on"
+)
+
+func resourceIBMPrivateDNSGLBMonitor() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMPrivateDNSGLBMonitorCreate,
+		Read:     resourceIBMPrivateDNSGLBMonitorRead,
+		Update:   resourceIBMPrivateDNSGLBMonitorUpdate,
+		Delete:   resourceIBMPrivateDNSGLBMonitorDelete,
+		Exists:   resourceIBMPrivateDNSGLBMonitorExists,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			pdnsGlbMonitorID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Monitor Id",
+			},
+
+			pdnsInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Instance Id",
+			},
+
+			pdnsGlbMonitorName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique identifier of a service instance.",
+			},
+
+			pdnsGlbMonitorDescription: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Descriptive text of the load balancer monitor",
+			},
+
+			pdnsGlbMonitorType: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "HTTP",
+				ValidateFunc: validateAllowedStringValue([]string{"HTTP", "HTTPS", "TCP"}),
+				Description:  "The protocol to use for the health check",
+			},
+
+			pdnsGlbMonitorPort: {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Port number to connect to for the health check",
+			},
+
+			pdnsGlbMonitorInterval: {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     60,
+				Description: "The interval between each health check",
+			},
+
+			pdnsGlbMonitorRetries: {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     1,
+				Description: "The number of retries to attempt in case of a timeout before marking the origin as unhealthy",
+			},
+
+			pdnsGlbMonitorTimeout: {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     5,
+				Description: "The timeout (in seconds) before marking the health check as failed",
+			},
+
+			pdnsGlbMonitorMethod: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"GET", "HEAD"}),
+				Description:  "The method to use for the health check",
+			},
+
+			pdnsGlbMonitorPath: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The endpoint path to health check against",
+			},
+
+			pdnsGlbMonitorHeaders: {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
```

Comment:
Description is missing for this attribute

Reply:
handled